packages/@aws-cdk/aws-iotevents/lib/input.ts

```diff
@@ -11,8 +12,66 @@ export interface IInput extends IResource {
    * @attribute
    */
   readonly inputName: string;
+
+  /**
+   * The ARN of the input
+   * @attribute
+   */
+  readonly inputArn: string;
+
+  /**
+   * Grant the indicated permissions on this input to the given IAM principal (Role/Group/User).
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param actions The set of actions to allow (i.e. "iotevents:BatchPutMessage")
+   */
+  grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant
+
+  /**
+   * Grant the putting message permission to the given IAM principal (Role/Group/User).
+   *
+   * @param grantee The principal (no-op if undefined)
+   */
+  grantPutMessage(grantee: iam.IGrantable): iam.Grant
 }
 
+
+abstract class InputBase extends Resource implements IInput {
+  /**
+   * @attribute
+   */
+  public abstract readonly inputName: string;
+  /**
+   * @attribute
+   */
+  public abstract readonly inputArn: string;
+
+  /**
+   * Grant the indicated permissions on this input to the given IAM principal (Role/Group/User).
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param actions The set of actions to allow (i.e. "iotevents:BatchPutMessage")
+   */
+  public grant(grantee: iam.IGrantable, ...actions: string[]) {
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [this.inputArn],
+      scope: this,
+    });
+  }
+
+  /**
+   * Grant the putting message permission to the given IAM principal (Role/Group/User).
+   *
+   * @param grantee The principal (no-op if undefined)
+   */
+  public grantPutMessage(grantee: iam.IGrantable) {
```

Comment:
Same here - put this above `grant()` (the method goes first, then the method(s) it uses in its implementation).